include/onnxruntime/core/framework/sparse_tensor.h

```diff
@@ -1,72 +1,317 @@
+  }
 
-  ~SparseTensor() = default;
+  SparseTensor();
+
+  ~SparseTensor();
 
   // For now, disallow all copy, assignment, and move.
-  ORT_DISALLOW_COPY_ASSIGNMENT_AND_MOVE(SparseTensor);
+  ORT_DISALLOW_COPY_AND_ASSIGNMENT(SparseTensor);
 
   // Returns the number of entries in the values tensor (aka "NNZ" or "number of nonzero values")
-  size_t NumValues() const { return static_cast<size_t>(values_.Shape().Size()); }
+  // For block sparse formats this may include some zeros in the blocks are considered non-zero.
+  size_t NumValues() const { return static_cast<size_t>(rep_->Values().Shape().Size()); }
 
   const Tensor& Values() const {
-    return values_;
+    return rep_->Values();
   }
 
-  const Tensor& Indices() const {
-    return indices_;
+  SparseTensor(SparseTensor&& o) noexcept {
+    *this = std::move(o);
   }
 
-  const TensorShape& Shape() const {
-    return shape_;
+  SparseTensor& operator=(SparseTensor&& o) noexcept;
+
+  /// <summary>
+  /// Returns SparseFormat that the instance currently holds
+  /// if the value returned in kUndefined, the instance is not populated
+  /// and any action except MutableRep<>() will fail
+  /// </summary>
+  /// <returns></returns>
+  SparseFormatFlags FormatFlags() const noexcept {
+    return format_flags_;
   }
 
-  Tensor& MutableValues() {
-    return values_;
+  /// <summary>
+  /// Tests if a specific flag set in the mask
+  /// </summary>
+  /// <param name="flag">flag to be tested</param>
+  /// <returns></returns>
+  bool IsFormatFlagSet(SparseFormatFlags flag) const noexcept {
+    return IsSet(format_flags_, flag);
   }
 
-  Tensor& MutableIndices() {
-    return indices_;
+  /// <summary>
+  /// Returns a would be dense_shape
+  /// This does not describe shapes of values of indices
+  /// and has no relation to
+  /// </summary>
+  /// <returns></returns>
+  const TensorShape& Shape() const noexcept {
+    return dense_shape_;
   }
 
-  //TensorShape& MutableShape() {
-  //  return shape_;
-  //}
+  /// <summary>
+  /// Calculates and returns how much this fully initialized SparseTensor data (would)
+  /// occupy in a contiguous allocation block, or, in fact, occupies if it owns the buffer.
+  /// </summary>
+  /// <returns></returns>
+  int64_t RequiredAllocationSize() const noexcept {
+    return rep_->RequiredAllocationSize();
+  }
+
+  /// <summary>
+  /// Returns Tensor element type enum.
+  /// Useful for type dispatching
+  /// </summary>
+  /// <returns></returns>
+  int32_t GetElementType() const {
+    return ml_data_type_->GetDataType();
+  }
+
+  /// <summary>
+  /// Return Element MLDataType
+  /// </summary>
+  /// <returns></returns>
+  MLDataType DataType() const noexcept {
+    return ml_data_type_;
+  }
+
+  /// <summary>
+  /// Test for string type
+  /// </summary>
+  /// <typeparam name="Rep"></typeparam>
+  /// <returns>true if tensor values are strings</returns>
+  bool IsDataTypeString() const {
```

Comment:
What's the reason for this being a separate method?

Reply:
Because it is widespread and it is similar to Tensor class.